pachli/pachli-android
# Add Mastodon Redirect support and "Open Link" share target

Very similar to https://github.com/tuskyapp/Tusky/pull/4194. Modified description below.

This PR adds auto-detect support for [Mastodon Redirect](https://github.com/zacharee/MastodonRedirect), so Pachli can be chosen for opening Mastodon links. It also adds an "Open Link" share target which uses the same logic path Mastodon Redirect does.

To test:
1. Download Mastodon Redirect from https://github.com/zacharee/MastodonRedirect.
2. Install Pachli from this PR's code.
3. Open and log into Pachli.
4. Find a post and copy its link. Paste the link somewhere it'll be clickable, such as in Telegram's "Saved messages" channel.
5. Remove Pachli from Recents.
6. Open Mastodon Redirect and select Pachli in the "Discovered" section.
7. Tap the pasted link. It should launch Pachli and then resolve to the post.
8. Back out to the main feed in Pachli and go back to the link, without removing Pachli from Recents this time. The link should open Pachli and resolve to the post.

To test the share target:
1. Install Pachli from this PR's code.
2. Find a post and copy its link into an app that supports sharing links, like Discord.
3. Share the link.
4. In the share sheet, choose Pachli's "Open Link" target. The previous/existing share target should also show as "Compose".
5. Pachli should open and resolve the link.
